fe/fe-core/src/main/java/org/apache/doris/nereids/hint/LeadingHint.java

```diff
@@ -124,6 +124,32 @@ public LeadingHint(String hintName, List<String> parameters, String originalStri
             }
             lastParameter = parameter;
         }
+        normalizeLevelList();
+    }
+
+    private void removeGap(int left, int right, int gap) {
+        for (int i = left; i <= right; i++) {
+            levellist.set(i, levellist.get(i) - (gap - 1));
+        }
+    }
+
+    // when leading extract levellist like 0 2 2 3 3 0, it could be reduced to 0 1 1 2 2 0
```

Comment:
please define what is the “gap”
it is better to copy the example in "Proposed Change" to illustrate the concept of gap

Reply:
get

Reply:
gap is like 0 to 2 in upper example, in this view we can see the represent of gap in levelList level